middleware/middleware_test.go

```diff
@@ -73,3 +73,32 @@ func TestMaxConcurrentRequestsMiddleware(t *testing.T) {
 		})
 	}
 }
+
+func TestContentTypeMiddleware(t *testing.T) {
+	nextHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	})
+	t.Run("test SetJsonContentType middleware", func(t *testing.T) {
+		expectedContentType := "application/json; charset=utf-8"
+		contentTypeMiddleware := middleware.SetJsonContentType()
+		handlerToTest := contentTypeMiddleware(nextHandler)
+		respRecorder := httptest.NewRecorder()
+		req := httptest.NewRequest("GET", "http://testing", nil)
+
+		handlerToTest.ServeHTTP(respRecorder, req)
+		receivedContentType := respRecorder.Header()["Content-Type"][0]
+		require.Equal(t, expectedContentType, receivedContentType, "actual content type different than expected.")
+	})
+
+	t.Run("test SetPlainContentType middleware", func(t *testing.T) {
+		expectedContentType := "text/plain; charset=utf-8"
+		contentTypeMiddleware := middleware.SetPlainContentType()
+		handlerToTest := contentTypeMiddleware(nextHandler)
+		respRecorder := httptest.NewRecorder()
+		req := httptest.NewRequest("GET", "http://testing", nil)
+
+		handlerToTest.ServeHTTP(respRecorder, req)
+		receivedContentType := respRecorder.Header()["Content-Type"][0]
+		require.Equal(t, expectedContentType, receivedContentType, "actual content type different than expected.")
```

Comment:
We could set a special header in the `next` handler and then you assert that that header is in the response.

I'm saying this because if you comment out line 80 (i.e. `w.WriteHeader(http.StatusOK)`) the test passes, so I think we could have a stronger guarantee that next is called.

The tests pass also if you comment out `next.ServeHTTP(w, r)` inside the middlewares.

```suggestion
		w.WriteHeader(http.StatusAccepted)
	})
	t.Run("test SetJsonContentType middleware", func(t *testing.T) {
		expectedContentType := "application/json; charset=utf-8"
		contentTypeMiddleware := middleware.SetJsonContentType()
		handlerToTest := contentTypeMiddleware(nextHandler)
		respRecorder := httptest.NewRecorder()
		req := httptest.NewRequest("GET", "http://testing", nil)

		handlerToTest.ServeHTTP(respRecorder, req)
		receivedContentType := respRecorder.Header()["Content-Type"][0]
		require.Equal(t, expectedContentType, receivedContentType, "actual content type different than expected.")
		require.Equal(t, http.StatusAccepted, respRecorder.Code, "actual response code different than expected.")
	})

	t.Run("test SetPlainContentType middleware", func(t *testing.T) {
		expectedContentType := "text/plain; charset=utf-8"
		contentTypeMiddleware := middleware.SetPlainContentType()
		handlerToTest := contentTypeMiddleware(nextHandler)
		respRecorder := httptest.NewRecorder()
		req := httptest.NewRequest("GET", "http://testing", nil)

		handlerToTest.ServeHTTP(respRecorder, req)
		receivedContentType := respRecorder.Header()["Content-Type"][0]
		require.Equal(t, expectedContentType, receivedContentType, "actual content type different than expected.")
		require.Equal(t, http.StatusAccepted, respRecorder.Code, "actual response code different than expected.")
```

Reply:
great point, we should definitely test that middleware is indeed performing the basic functionality that is is supposed to perform, i.e. calling the next function.

Modified the test to return custom statusCode in the next handler and then asserted for it.